docs/ABI.rst

```diff
@@ -909,92 +936,120 @@ constraints on the extension are mangled in its generic signature.
 When mangling the context of a local entity within a constructor or
 destructor, the non-allocating or non-deallocating variant is used.
 
+::
+
+  module ::= substitution                    // other substitution
+  module ::= identifier                      // module name
+  module ::= known-module                    // abbreviation
+
+  known-module ::= 's'                       // Swift
+  known-module ::= 'SC'                      // C
+  known-module ::= 'So'                      // Objective-C
+
+The Objective-C module is used as the context for mangling Objective-C
+classes as ``<type>``\ s.
+
+
 Types
 ~~~~~
 
 ::
 
+  nominal-type ::= substitution
+  nominal-type ::= context decl-name nominal-type-kind
+
+  nominal-type-kind ::= 'C'                  // class
+  nominal-type-kind ::= 'O'                  // enum
+  nominal-type-kind ::= 'V'                  // struct
```

Comment:
If we're going to redo the mangling, we should do away with differentiating nominal type kinds, IMO.